pandas/core/arrays/datetimes.py

```diff
@@ -322,9 +324,20 @@ def __init__(self, values, dtype=_NS_DTYPE, freq=None, copy=False):
         self._dtype = dtype
         self._freq = freq
 
+        if inferred_freq is None and freq is not None:
+            type(self)._validate_frequency(self, freq)
+
     @classmethod
-    def _simple_new(cls, values, freq=None, dtype=None):
-        return cls(values, freq=freq, dtype=dtype)
+    def _simple_new(cls, values, freq=None, dtype=_NS_DTYPE):
+        assert isinstance(values, np.ndarray)
+        if values.dtype == 'i8':
```

Comment:
umm, shouldn't these *always* be i8?

Reply:
No.  they can be datetime64[ns]

Reply:
*really*? this is really really strange to do this, we should have a single implementation.

Reply:
Really really.  I don’t have the bandwidth to address this concern ATM.  Pls open an issue if you feel strongly about it.